packages/ra-core/src/auth/usePermissions.ts

```diff
@@ -42,25 +32,23 @@ const emptyParams = {};
  *     };
  */
 const usePermissions = <Permissions = any, Error = any>(
-    params = emptyParams
-): State<Permissions, Error> => {
-    const [state, setState] = useSafeSetState<State<Permissions, Error>>({
-        isLoading: true,
-    });
+    params: UseQueryOptions<Permissions, Error> = {
+        refetchOnMount: false,
+    }
+) => {
     const getPermissions = useGetPermissions();
-    useEffect(() => {
-        getPermissions(params)
-            .then(permissions => {
-                setState({ isLoading: false, permissions });
-            })
-            .catch(error => {
-                setState({
-                    isLoading: false,
-                    error,
-                });
-            });
-    }, [getPermissions, params, setState]);
-    return state;
+
+    const { data, isLoading, error } = useQuery(
+        ['ra-Permissions'],
+        getPermissions,
```

Comment:
This is a breaking change, as we used to pass in the params

Reply:
also passed now